lightning/src/ln/channelmanager.rs

```diff
@@ -4404,3 +4407,154 @@ mod tests {
 		}
 	}
 }
+
+#[cfg(all(any(test, feature = "_test_utils"), feature = "unstable"))]
+pub mod bench {
+	use chain::Listen;
+	use chain::chainmonitor::ChainMonitor;
+	use chain::channelmonitor::Persist;
+	use chain::keysinterface::{KeysManager, InMemorySigner};
+	use chain::transaction::OutPoint;
+	use ln::channelmanager::{ChainParameters, ChannelManager, PaymentHash, PaymentPreimage};
+	use ln::features::InitFeatures;
+	use ln::functional_test_utils::*;
+	use ln::msgs::ChannelMessageHandler;
+	use routing::network_graph::NetworkGraph;
+	use routing::router::get_route;
+	use util::test_utils;
+	use util::config::UserConfig;
+	use util::events::{Event, EventsProvider, MessageSendEvent, MessageSendEventsProvider};
+
+	use bitcoin::hashes::Hash;
+	use bitcoin::hashes::sha256::Hash as Sha256;
+	use bitcoin::{Block, BlockHeader, Transaction, TxOut};
+
+	use std::sync::Mutex;
+
+	use test::Bencher;
+
+	struct NodeHolder<'a, P: Persist<InMemorySigner>> {
+		node: &'a ChannelManager<InMemorySigner,
```

Comment:
I'd be curious to compare this to an `Arc`'d version (i.e. using arcs instead of refs)

Reply:
I'd be astounded if it made a difference - we aren't actually cloning the `Arc`s anywhere that I know of here and our runtime is dominated by cryptographic operations. Non-Arc objects are nice mostly because we'd like to eventually support non-threaded applications, so we don't need trait implementations to implement `Sync+Send`.